locator/topology.hh

```diff
@@ -167,7 +172,7 @@ public:
 
         bool operator==(const config&) const = default;
     };
-    topology(config cfg);
+    topology(config cfg, key_kind k);
```

Comment:
why not make it part of `topology::config`?

Reply:
it's temporary anyway